File: gcode/src/main/java/org/qw3rtrun/p3d/g/event/TemperatureReported.java
package org.qw3rtrun.p3d.g.event;

import org.qw3rtrun.p3d.core.msg.TemperatureReport;
import org.qw3rtrun.p3d.core.msg.TemperatureReportedEvent;

public record TemperatureReported(TemperatureReport hotend,
                                  TemperatureReport bed) implements TemperatureReportedEvent {
}

File: gcode/src/main/kotlin/org/qw3rtrun/p3d/g/marlin/event/TemperatureReported.kt
package org.qw3rtrun.p3d.g.marlin.event

import org.qw3rtrun.p3d.core.msg.TemperatureReport
import org.qw3rtrun.p3d.core.msg.TemperatureReportedEvent

data class TemperatureReported(
    @get:JvmName("getHotend") val hotend: TemperatureReport?,
    @get:JvmName("getBed") val bed: TemperatureReport?
) : TemperatureReportedEvent {
    override fun hotend(): TemperatureReport? = hotend
    override fun bed(): TemperatureReport? = bed
}
